### PR TITLE
[5.1] Header list field

### PR DESCRIPTION
**Pull request by Razzo1987**
Improving @brianteeman PR #41122 to extend to field lists.
Thanks to @alikon for the support.

### Summary of Changes
 Extending the possibility of using the header to list fields as well.
So moving some code from `SqlField.php` to `ListField.php`.

### Testing Instructions
1) Create a file of type List
2) Create a file of type SQL

3) Create a XML field and add header option

### Actual result BEFORE applying this Pull Request
1) you cannot specify the header, so the first value in the list is always selected when creating an article (even if the field is indicated as not required)
2) You can specify the header to obviate the problem described in case 1.

### Expected result AFTER applying this Pull Request
1) You may now specify the header
2) The SQL field still works

3) now you can add the header from XML too

### Link to documentations
Please select:
- [X] Documentation link for docs.joomla.org: [List_form_field_type](https://docs.joomla.org/List_form_field_type)
- [ ] No documentation changes for docs.joomla.org needed

- [ ] Pull Request link for manual.joomla.org: <link>
- [X] No documentation changes for manual.joomla.org needed

Note:
Eg of SQL query for SQL Field (you need to create some articles for example from sample data):
```
SELECT #__content.id AS value, #__content.title AS text
FROM #__content
```

Eg of XML list field:
```
	<field
		name="test"
		type="list"
		label="test"
		validate="options"
		header="Please select your option"
		>
		<option value="1">JYES</option>
		<option value="0">JNO</option>
	</field>
```

### CODE REVIEW

File: libraries/src/Form/Field/ListField.php [hunk 1]
```diff
@@ -232,4 +247,30 @@ public function __get($name)
...
+     * @see     FormField::setup()
+     * @since   __DEPLOY_VERSION__
+     */
+    public function setup(\SimpleXMLElement $element, $value, $group = null)
+    {
+        $return = parent::setup($element, $value, $group);
+
+        if ($return) {
+            // Check if its using the old way
+            $this->header     = (string) $this->element['header'] ?: false;
```
Comment (Quy) [inline libraries/src/Form/Field/ListField.php]:
```suggestion
        if ($return) {
            // Check if it's using the old way
            $this->header = (string) $this->element['header'] ?: false;
```

File: plugins/fields/list/params/list.xml [hunk 2]
```diff
@@ -2,6 +2,14 @@
 <form>
 	<fields name="fieldparams">
 		<fieldset name="fieldparams">
+			<field
+					name="header"
+					type="text"
+					label="PLG_FIELDS_LIST_PARAMS_HEADER_LABEL"
+					description="PLG_FIELDS_LIST_PARAMS_HEADER_DESC"
+					filter="string"
+				/>
```
Comment (Quy) [inline plugins/fields/list/params/list.xml]:
```suggestion
				name="header"
				type="text"
				label="PLG_FIELDS_LIST_PARAMS_HEADER_LABEL"
				description="PLG_FIELDS_LIST_PARAMS_HEADER_DESC"
				filter="string"
			/>
```

**Comment by faustonenci** [comment]
I have tested this item :white_check_mark: successfully on 9c1290799cd18e1008de24ffd2ef279aab25fcd2<br /><br />good works<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/42736">issues.joomla.org/tracker/joomla-cms/42736</a>.</sub>

**Comment by walturbo** [comment]
I have tested this item :white_check_mark: successfully on 9c1290799cd18e1008de24ffd2ef279aab25fcd2<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/42736">issues.joomla.org/tracker/joomla-cms/42736</a>.</sub>

**Comment by alikon** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/42736">issues.joomla.org/tracker/joomla-cms/42736</a>.</sub>

**Comment by adj9** [comment]
I have not tested this item.<br /><br />its ok<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/42736">issues.joomla.org/tracker/joomla-cms/42736</a>.</sub>

**Comment by bembelimen** [comment]
I see the benefit of this PR, but there should be the following considerations:
- "Header" is for sure the wrong naming for such a parameter
- There should be a "show on" function to depend on the "required" field
- Consider a language string as default?

**Comment by bembelimen** [comment]
Ok, @LadySolveig fought for this PR and convinced me, that we should merge it, so here we go!

**Comment by bembelimen** [comment]
Thx

**Pull request merged**